### PR TITLE
Support chacha20-poly1305@openssh.com.

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/tmds/Tmds.Ssh/issues/214.

@jborean93 fyi

### CODE REVIEW

**Comment by tmds** [comment]
I currently put `chacha20-poly1305@openssh.com` first in `SupportedEncryptionAlgorithms` so all (`SshClientSettings`-based) tests are now running with this algorithm.

I will add new tests to this PR which test the ciphers separately.

File: src/Tmds.Ssh/SshClientSettings.Defaults.cs [hunk 1]
```diff
@@ -63,6 +65,40 @@ private static IReadOnlyList<Credential> CreateDefaultCredentials()
         return credentials.AsReadOnly();
     }
 
+    private static List<Name> CreatePreferredEncryptionAlgorithms()
+    {
+        // Prefer AesGcm over ChaCha20Poly when the platform has AES instructions.
```
Comment (jborean93) [inline src/Tmds.Ssh/SshClientSettings.Defaults.cs:70]: Do we do this as AES with hardware acceleration is faster or is this what OpenSSH does as well?

File: src/Tmds.Ssh/SshClientSettings.Defaults.cs [hunk 2]
```diff
@@ -63,6 +65,40 @@ private static IReadOnlyList<Credential> CreateDefaultCredentials()
...
+        bool addAesGcm = AesGcm.IsSupported;
+        bool hasAesInstructions =  System.Runtime.Intrinsics.X86.Aes.X64.IsSupported ||
```
Comment (jborean93) [inline src/Tmds.Ssh/SshClientSettings.Defaults.cs:72]: Shouldn't we be checking based on the process architecture and not all of them?

**Review by jborean93 — COMMENTED** [review]
I can't really talk about the encryption code as I don't really understand chacha20poly1305 but the rest seems ok to me.

File: src/Tmds.Ssh/SshClientSettings.Defaults.cs (hunk 1 above)
Comment (tmds) [reply src/Tmds.Ssh/SshClientSettings.Defaults.cs:70]: This is for ordering from fastest to slowest.

OpenSSH doesn't try to be smart. It has a fixed order.
The system config files enable the administrator to re-order the algorithms.

The `SshClient` can be created so it uses the OpenSSH config files (and follows their configuration).

File: src/Tmds.Ssh/SshClientSettings.Defaults.cs (hunk 2 above)
Comment (tmds) [reply src/Tmds.Ssh/SshClientSettings.Defaults.cs:72]: I think we can just check all of them. When we're checking for a different architecture, the `IsSupported` will return `false`.

**Pull request merged**